docs/content/latest/api/ycql/batch.md

```diff
@@ -0,0 +1,173 @@
+b.add(f"UPDATE test_rs_batch SET v2='z' WHERE h=3 AND r=3 IF v2='z'  RETURNS STATUS AS ROW;")
+b.add(f"DELETE FROM test_rs_batch WHERE h=2 AND r=2 IF EXISTS RETURNS STATUS AS ROW;")
+
+# Execute the batch operation.
+result = session.execute(b, trace=True)
+
+# Print status for each DML operation
+for row in result:
+    print(row)
+```
+
+The output generated is:
+
+```python
+Row(applied=True, message=None, h=None, r=None, v1=None, v2=None)
+Row(applied=False, message=None, h=3, r=3, v1=None, v2='b')
+Row(applied=False, message=None, h=None, r=None, v1=None, v2=None)
+```
+
+## Row Status
+
+When executing a batch in YCQL, the protocol just allows returning one error/return status.
+
+So, if one statement fails with an error or, for conditional DMLs, some are not applied because of failing the IF condition, 
+the driver/application cannot accurately identify the relevant statement(s), it will just receive one general error/return-status for the whole batch.
+
+Therefore, it is not trivial/possible for an application to react to such failures appropriately (e.g. retry, abort, 
+change some parameters, for either entire batch or just the relevant statements etc).
+
+The `RETURNS STATUS AS ROW` feature addresses this limitation.
+
+
+If used, the (write) statement will return it's status (whether applied, unapplied, or errored-out with a message) as a regular CQL 
+row that the application can inspect and decide what to do.
+
+In the case of a batch (which is its intended use-case), it is required that either none or all statements use `RETURNS STATUS AS ROW`.
+
+
+This is so that when executing n statements in a batch with `RETURN STATUS AS ROW` it will always return n rows (in the same order as the statements) 
+and the app can easily inspect the result.
+
+Similarly, it is required that `RETURN STATUS AS ROW` is used for batches containing conditional DMLs.
+
+This is because, for conditional DMLs (which are normally not allowed in batches), any subset of them could fail due to 
+their `IF` condition so only returning rows for them would make it impossible to identify which ones actually failed.
+
+To distinguish between the two not-applied cases (error vs condition is false) there is an (error) `message` column in 
+the return row that will be null for not-applied and filled-in for errors.
+
+Conversely, there will be one column for each table column which will be `null` for errors but filled-in for not-applied 
+(justifying the decision to not apply).
+
+For instance:
+
+1. Set up a simple table:
+```cassandraql
+cqlsh:sample> CREATE TABLE test(h INT, r INT, v LIST<INT>, PRIMARY KEY(h,r)) WITH transactions={'enabled': true};
+cqlsh:sample> INSERT INTO test(h,r,v) VALUES (1,1,[1,2]);
+```
+
+2. Unapplied update when `IF` condition is false:
+```cassandraql
+cqlsh:sample> UPDATE test SET v[2] = 4 WHERE h = 1 AND r = 1 IF v[1] = 3 RETURNS STATUS AS ROW;
```

Comment:
does the copy button work fine for these cases?
Does it not also copy the output? Also the highlighting does not seem to work.
Same for the other cases, check that all Copy's on the table extracts (only) the relevant CQL text.